### ISSUE TITLE
This browser or app may not be secure.

### ISSUE THREAD

**Issue by chris0piper** [linouk23/youtube_uploader_selenium#78]
Any ideas on how to fix this? I've tried copying my firefox profile folder from my local comp into this scripts profile folder but no success

**Comment by chenaaron3** [comment]
Are you using the correct firefox profile? It works for me when I specify a profile path.

```
# export PROFILE_PATH="C:\Users\chena\AppData\Roaming\Mozilla\Firefox\Profiles\1cyrfe0a.default-release-1676168063686"
uploader = YouTubeUploader(video_path, metadata_path, profile_path=os.getenv('PROFILE_PATH'))  
was_video_uploaded, video_id = uploader.upload()
```

**Comment by arkilis** [comment]
@chenaaron3 Had a try with your method but still error:

```
TypeError: Firefox.__init__() got an unexpected keyword argument 'profile_path'
```

**Comment by mqxmm** [comment]
> @chenaaron3 Had a try with your method but still error:
> 
> ```
> TypeError: Firefox.__init__() got an unexpected keyword argument 'profile_path'
> ```

up up

**Comment by FujiwaraChoki** [comment]
> @chenaaron3 Had a try with your method but still error:
> 
> ```
> TypeError: Firefox.__init__() got an unexpected keyword argument 'profile_path'
> ```

Same error

**Comment by cindylaii** [comment]
> Are you using the correct firefox profile? It works for me when I specify a profile path.
> 
> ```
> # export PROFILE_PATH="C:\Users\chena\AppData\Roaming\Mozilla\Firefox\Profiles\1cyrfe0a.default-release-1676168063686"
> uploader = YouTubeUploader(video_path, metadata_path, profile_path=os.getenv('PROFILE_PATH'))  
> was_video_uploaded, video_id = uploader.upload()
> ```

This method solved my problem, and I can now log into YouTube.

**Comment by modpotato** [comment]
> Are you using the correct firefox profile? It works for me when I specify a profile path.
> 
> ```
> # export PROFILE_PATH="C:\Users\chena\AppData\Roaming\Mozilla\Firefox\Profiles\1cyrfe0a.default-release-1676168063686"
> uploader = YouTubeUploader(video_path, metadata_path, profile_path=os.getenv('PROFILE_PATH'))  
> was_video_uploaded, video_id = uploader.upload()
> ```

this doesnt work for me and i constantly get An error occurred during upload: YouTubeUploader.__init__() got an unexpected keyword argument 'profile_path' did you modify your version?

edit: i was using the pypi latest version which frankly isnt updated to the most recent version of this github repo, i just manually replaced the code in my venv and it may work now

**Comment by cindylaii** [comment]
> > Are you using the correct firefox profile? It works for me when I specify a profile path.
> > ```
> > # export PROFILE_PATH="C:\Users\chena\AppData\Roaming\Mozilla\Firefox\Profiles\1cyrfe0a.default-release-1676168063686"
> > uploader = YouTubeUploader(video_path, metadata_path, profile_path=os.getenv('PROFILE_PATH'))  
> > was_video_uploaded, video_id = uploader.upload()
> > ```
> 
> this doesnt work for me and i constantly get An error occurred during upload: YouTubeUploader.**init**() got an unexpected keyword argument 'profile_path' did you modify your version?
> 
> edit: i was using the pypi latest version which frankly isnt updated to the most recent version of this github repo, i just manually replaced the code in my venv and it may work now

My previous question was that I had never used Firefox's profile feature before, so I couldn't understand the instructions in the installation guide. Later, I searched online about profiles and learned that it's a Firefox feature for customizing profiles. You first create a Firefox profile, then fill in the profile path in the program. I've been using this program for several months now, and the main part of the program works fine.

**Comment by modpotato** [comment]
> > > Are you using the correct firefox profile? It works for me when I specify a profile path.
> > > ```
> > > # export PROFILE_PATH="C:\Users\chena\AppData\Roaming\Mozilla\Firefox\Profiles\1cyrfe0a.default-release-1676168063686"
> > > uploader = YouTubeUploader(video_path, metadata_path, profile_path=os.getenv('PROFILE_PATH'))  
> > > was_video_uploaded, video_id = uploader.upload()
> > > ```
> > 
> > 
> > this doesnt work for me and i constantly get An error occurred during upload: YouTubeUploader.**init**() got an unexpected keyword argument 'profile_path' did you modify your version?
> > edit: i was using the pypi latest version which frankly isnt updated to the most recent version of this github repo, i just manually replaced the code in my venv and it may work now
> 
> My previous question was that I had never used Firefox's profile feature before, so I couldn't understand the instructions in the installation guide. Later, I searched online about profiles and learned that it's a Firefox feature for customizing profiles. You first create a Firefox profile, then fill in the profile path in the program. I've been using this program for several months now, and the main part of the program works fine.

while that may be true, the code in this repo uses a very old method of setting the profile which isnt recommended anymore for one reason or another

i ended up replacing it with the following
```python
        options = Options()
        options.add_argument("-profile")
        options.add_argument(profile_path)
```

**Comment by cindylaii** [comment]
> ```python
> profile_path
> ```

Can you tell me the string of the profile_path?

**Comment by modpotato** [comment]
> > ```python
> > profile_path
> > ```
> 
> Can you tell me the string of the profile_path?

no, its different for each person but you would find it in about:profiles 90% of the time

**Comment by cindylaii** [comment]
> no, its different for each person but you would find it in about:profiles 90% of the time

I just want to confirm that you have added the correct Firefox profile path. I think you should be able to use it normally.
